src/main/java/me/botsko/prism/serializers/items/ItemStackSerializer.java

```diff
@@ -0,0 +1,295 @@
+            data.bannerMeta = stringyPatterns;
+        }
+        return data;
+    }
+
+    private static void applyFireWorkMetaToActionData(ItemMeta meta, ItemStackSerializer data) {
+        final FireworkEffectMeta fireworkMeta = (FireworkEffectMeta) meta;
+        if (fireworkMeta.hasEffect()) {
+            final FireworkEffect effect = fireworkMeta.getEffect();
+            if (effect != null) {
+                if (!effect.getColors().isEmpty()) {
+                    final int[] effectColors = new int[effect.getColors().size()];
+                    int i = 0;
+                    for (final Color effectColor : effect.getColors()) {
+                        effectColors[i] = effectColor.asRGB();
+                        i++;
+                    }
+                    data.effectColors = effectColors;
+                }
+
+                if (!effect.getFadeColors().isEmpty()) {
+                    final int[] fadeColors = new int[effect.getColors().size()];
+                    final int i = 0;
+                    for (final Color fadeColor : effect.getFadeColors()) {
+                        fadeColors[i] = fadeColor.asRGB();
+                    }
+                    data.fadeColors = fadeColors;
+                }
+                if (effect.hasFlicker()) {
+                    data.hasFlicker = true;
+                }
+                if (effect.hasTrail()) {
+                    data.hasTrail = true;
+                }
+            }
+        }
+    }
+
+    public ItemStack toBukkit() {
+        Material m = Material.matchMaterial(material);
+        if (m == null)
+            return null;
+        ItemStack item = new ItemStack(m, amt);
+
+        ItemUtils.setItemDamage(item, durability);
+
+        // Restore enchantment
+        if (enchs != null && enchs.length > 0) {
+            for (final String ench : enchs) {
+                final String[] enchArgs = ench.split(":");
+                Enchantment enchantment = Enchantment.getByKey(NamespacedKey.minecraft(enchArgs[0]));
+
+                // Restore book enchantment
+                if (enchantment != null) {
+                    if (item.getType() == Material.ENCHANTED_BOOK) {
+                        final EnchantmentStorageMeta bookEnchantments = (EnchantmentStorageMeta) item.getItemMeta();
+                        if (bookEnchantments != null) {
+                            bookEnchantments.addStoredEnchant(enchantment, Integer.parseInt(enchArgs[1]), false);
+                            item.setItemMeta(bookEnchantments);
+                        }
+                    } else {
+                        item.addUnsafeEnchantment(enchantment, Integer.parseInt(enchArgs[1]));
+                    }
+                }
+            }
+        }
+
+        ItemMeta meta = item.getItemMeta();
+        if (name != null) {
+            if (meta != null) {
```

Comment:
Codacy found an issue: [These nested if statements could be combined](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=6368169)